d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -1159,6 +1203,12 @@ public DarkClusterConfigMap getDarkClusterConfigMap(String clusterName) throws S
     }
     catch (ExecutionException | TimeoutException | IllegalStateException | InterruptedException e )
     {
+      if(e instanceof TimeoutException){
+        if(_state.getClusterProperties(clusterName) != null && _state.getClusterProperties(clusterName).getProperty() != null){
+          ClusterProperties clusterProperties = _state.getClusterProperties(clusterName).getProperty();
+          return clusterProperties != null ? clusterProperties.accessDarkClusters() : new DarkClusterConfigMap();
+        }
```

Comment:
line 1224 to 1226 can be put into a private helper function, to be shared to call here and there.

Same pattern applies to getClusterCount